app/controllers/stats_controller.rb

```diff
@@ -14,19 +16,14 @@ def index
   def show
     report = Report.new report_type_id: params[:id]
 
-    if report.xlsx?
-      report_data = report.run
-
-      axlsx = create_spreadsheet(report_data)
-
-      send_data axlsx.to_stream.read,
-                filename: report.report_type.filename('xlsx'),
-                disposition: :attachment,
-                type: SPREADSHEET_CONTENT_TYPE
+    report.run_and_update!(user: current_user)
+    if report.etl?
```

Comment:
I think we may of spoken about this yesterday - just wondering if `etl?` is the best method name here when I think it is really checking that the report is `processedAsBackgroundJob?`